components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/admin/impl/DefaultConsentAdminHandler.java

```diff
@@ -0,0 +1,335 @@
+
+            JSONArray consentHistory = new JSONArray();
+            for (Map.Entry<String, ConsentHistoryResource> result : results.entrySet()) {
+                JSONObject consentResourceJSON = new JSONObject();
+                ConsentHistoryResource consentHistoryResource = result.getValue();
+                DetailedConsentResource detailedConsentHistory = consentHistoryResource.getDetailedConsentResource();
+                consentResourceJSON.put(ConsentExtensionConstants.HISTORY_ID, result.getKey());
+                consentResourceJSON.put(ConsentExtensionConstants.AMENDED_REASON,
+                        consentHistoryResource.getReason());
+                consentResourceJSON.put(ConsentExtensionConstants.AMENDED_TIME,
+                        detailedConsentHistory.getUpdatedTime());
+                consentResourceJSON.put(ConsentExtensionConstants.CONSENT_DATA,
+                        ConsentAdminUtils.detailedConsentToJSON(detailedConsentHistory));
+                consentHistory.put(consentResourceJSON);
+            }
+            response.put(ConsentExtensionConstants.CC_CONSENT_ID, consentId);
+            response.put(ConsentExtensionConstants.CURRENT_CONSENT,
+                    ConsentAdminUtils.detailedConsentToJSON(ConsentExtensionsDataHolder.getInstance()
+                            .getConsentCoreService().getDetailedConsent(consentId)));
+            response.put(ConsentExtensionConstants.AMENDMENT_HISTORY, consentHistory);
+            count = consentHistory.length();
+        } catch (ConsentManagementException e) {
+            log.error("Error while retrieving consent amendment history data", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+
+        JSONObject metadata = new JSONObject();
+        metadata.put(ConsentExtensionConstants.AMENDMENT_COUNT, count);
+        response.put(ConsentExtensionConstants.METADATA, metadata);
+        consentAdminData.setResponseStatus(ResponseStatus.OK);
+        consentAdminData.setResponsePayload(response);
+    }
+
+    @Override
+    public void handleConsentStatusAuditSearch(ConsentAdminData consentAdminData) throws ConsentException {
+
+        JSONObject response = new JSONObject();
+        Integer limit = null;
+        Integer offset = null;
+
+        Map queryParams = consentAdminData.getQueryParams();
+
+        ArrayList<String> consentIDs =  ConsentAdminUtils.getArrayListFromQueryParam(ConsentAdminUtils
+                .validateAndGetQueryParam(queryParams, ConsentExtensionConstants.CONSENT_IDS));
+        try {
+            limit = ConsentAdminUtils.getIntFromQueryParam(ConsentAdminUtils
+                    .validateAndGetQueryParam(queryParams, ConsentExtensionConstants.LIMIT));
+        } catch (NumberFormatException e) {
+            log.error("Number format incorrect in search for parameter limit. Ignoring parameter");
```

Comment:
I think this shouldn't be an error log.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/9/commits/7640a3fd126e5e00ce693961fdc6c5cd28816bb9